File: app/src/main/java/com/revlogix/app/data/local/CustomPart.kt
package com.revlogix.app.data.local

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "custom_parts",
    foreignKeys = [
        ForeignKey(
            entity = Vehicle::class,
            parentColumns = ["vehicleId"],
            childColumns = ["vehicleId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class CustomPart(
    @PrimaryKey(autoGenerate = true)
    val partId: Int = 0,
    val vehicleId: Int,
    val category: String,
    val partName: String,
    val brand: String,
    val partNumber: String,
    val serialNumber: String,
    val technicalSpecs: String,
    val vendor: String,
    val purchaseDate: Long?,
    val installDate: Long?,
    val cost: Double,
    val notes: String = ""
)

File: RevLogixApi/Models/CustomPart.cs
using System.ComponentModel.DataAnnotations;

namespace RevLogixApi.Models;

public class CustomPart
{
    [Key]
    public int PartId { get; set; }
    public int VehicleId { get; set; }
    public string Category { get; set; } = "";
    public string PartName { get; set; } = "";
    public string Brand { get; set; } = "";
    public string PartNumber { get; set; } = "";
    public string SerialNumber { get; set; } = "";
    public string TechnicalSpecs { get; set; } = "";
    public string Vendor { get; set; } = "";
    public double Cost { get; set; }
}